inst/templates/tidy_contributing.md

```diff
@@ -0,0 +1,147 @@
+# Contributing to {{{package}}}
+
+-   [Prerequisites](#prerequisites)
+-   [Package mechanics](#package-mechanics)
+    -   [Fork, clone, branch](#fork-clone-branch)
+    -   [Check](#check)
+    -   [Style](#style)
+    -   [Document](#document)
+    -   [Test](#test)
+    -   [NEWS](#news)
+    -   [Re-check](#re-check)
+    -   [Commit](#commit)
+-   [Push and pull](#push-and-pull)
+    -   [Review, revise, repeat](#review-revise-repeat)
+-   [Resources](#resources)
+-   [Code of Conduct](#code-of-conduct)
+
+This explains how to propose a change to {{{package}}} via a pull request using
+Git and Github. 
+
+For more general info about contributing to the tidyverse, see the 
+[Resources](#resources) at the end of this document.
+
+## Prerequisites
+
+Before you do a pull request, you should always file an issue and make sure
+someone from the tidyverse team agrees that it’s a problem, and is happy with
+your basic proposal for fixing it. If you’ve found a bug, first create a minimal
+[reprex](https://www.tidyverse.org/help/#reprex).
+
+## Package mechanics
+
+### Fork, clone, branch
+
+The first thing you'll need to do is to fork the 
+[{{{package}}} GitHub repo](https://github.com/tidyverse/{{{package}}}), and 
+then clone it locally. We recommend that you create a branch for each PR.
+
+### Check
+
+Before changing anything, make sure the package still passes `R CMD check`
+locally for you. You'll do this again _after_ you've made changes, but it's best
+to establish a baseline.
+
+``` r
+devtools::check()
+```
+
+### Style
+
+Match the existing code style. This means you should follow the tidyverse 
+[style guide](http://style.tidyverse.org). Use the 
+[styler](https://CRAN.R-project.org/package=styler) package to apply the style 
+guide automatically.
+
+
+Be careful to only make style changes to the code you are contributing. If you
+find that there is a lot of code that doesn't meet the style guide, it would be
+better to file an issue or a separate PR to fix that first.
+
+### Document
+
+We use [roxygen2](https://cran.r-project.org/package=roxygen2),
+specifically with the [Markdown syntax](https://cran.r-project.org/web/packages/roxygen2/vignettes/markdown.html),
+to create `NAMESPACE` and all `.Rd` files. All edits to documentation
+should be done in roxygen comments above the associated function or
+object.
+
+``` r
+devtools::document()
+```
```

Comment:
Unlike `devtools::check()`, it's not immediately clear what `devtools::document()` actually does. Possible wording:

All edits ... Then use `devtools::document()` to rebuild the `NAMESPACE` and `.Rd` files.